docs/wiki/EL/el-architecture.md

```diff
@@ -173,12 +173,39 @@ At its core, Geth runs the Ethereum Virtual Machine (EVM), which processes smart
 
-Because it's inefficient to validate block by block and transaction by transaction since the genesis, EL clients employ other strategies to securely sync the tip of the chain, e.g. [snap sync](https://github.com/ethereum/devp2p/blob/master/caps/snap.md?plain=1#L1)
+Let's look at how the flows of both **full sync** and **snap sync** work.
 
+### Full Sync
+Full Sync trades absolute trustlessness for time and resources. You replay every block and transaction from genesis to tip, rebuilding the state trie step by step:
+1. Use `GetBlockHeaders` over the `eth/*` protocol to download every block header since genesis.
+2. Use `GetBlockBodies` to retrieve every block’s transactions and uncles.
+3. Sequentially execute each transaction in EVM order, updating the local trie at each block.
+4. Confirm that the local state trie’s root matches the tip’s root. Every state transition has been verified.
+
+This method guarantees maximal security but can take days on mainnet and consumes significant CPU, disk, and network resources.
+
+### Snap Sync
+Snap Sync reconstructs the pivot block's state by fetching only the trie leaves (accounts and storage slots) plus Merkle proofs, then separately downloading any needed contract bytecode, and finally rebuilding the tries locally:
+1. Choose a recently finalized block as your pivot block.
+2. Over eth/*, request that block’s header via `GetBlockHeaders` to learn its stateRoot. 
+3. Fetch headers up to the pivot via `eth/*`, so you know which state root to target.
+4. Download the leaves of the pivot block's world state trie and account storage trie in chunks.
+   - **Accounts**: uses `GetAccountRange` to pull contiguous world state trie leaf values.
+   - **Storage**: uses `GetStorageRanges` to pull consecutive storage slot leaf nodes for each account.
+5. Send `GetByteCodes` for every code hash found in the account bodies to get back the contract code for that account
+6. Locally insert every fetched leaf into a fresh trie, verifying each batch against the pivot root via Merkle proofs.
+
+After step 6, we have a snapshot of the pivot block's state. However, the blockchain is also progressing at the same time and invalidating some of the regenerated state data. This means it is also necessary to have a healing phase where errors in the state are fixed.  During healing, the client walks the reconstructed trie and verifies that it hashes to the pivot block’s stateRoot. Any missing or inconsistent trie nodes or bytecode blobs are fetched using targeted snap requests. Healing ensures the final state is complete and cryptographically valid, even if parts of the snapshot were stale, incomplete, or assembled from mixed sources.  
```

Comment:
One small thing I would highlight here is that snapshot has a specific database like format, different from state merkle

Reply:
no doubt, added this context in https://github.com/eth-protocol-fellows/protocol-studies/pull/437/commits/2bde312198c716340ca067a292936d77415c1ae6